src/class_sqlThread.py

```diff
@@ -635,8 +523,45 @@ def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-s
                 helper_sql.sqlReturnQueue.put((self.cur.fetchall(), rowcount))
                 # helper_sql.sqlSubmitQueue.task_done()
 
+    def earlier_setting_version(self, settingsversion):
```

Comment:
move to parent

Reply:
Done

Reply:
different name, to show it's related to `BMConfigParser` version rather than sql version

Reply:
Change name to ```upgrade_config_parser_setting_version```